src/util/WalletBackendAdapter.ts

```diff
@@ -210,13 +291,29 @@ export const makeWalletBridgeFromFollowers = (
     }
     const currentEl: ValueFollowerElement<CurrentWalletRecord> = latest.value;
     const wallet = currentEl.value;
-    console.log('wallet current', wallet);
+    console.debug('wallet current', wallet);
+
+    const agoricBrands = await fetchAgoricBrands();
+    assert(agoricBrands, 'Failed to fetch agoric brands');
+
     for (const purse of wallet.purses) {
-      console.debug('registering purse', purse);
-      const brandDescriptor = wallet.brands.find(
-        bd => purse.brand === bd.brand,
-      );
-      assert(brandDescriptor, `missing descriptor for brand ${purse.brand}`);
+      // We only care about the zoe invite purse, which has asset kind 'set'.
+      // Non 'set' amounts need to be fetched from vbank to know their
+      // decimalPlaces.
+      //
+      // If we have add non 'set' amount purses that aren't in the vbank, it's
+      // not currently possible to read their decimalPlaces, so this code
+      // will need updating.
+      if (
+        !Array.isArray(purse.balance.value) ||
+        !agoricBrands.has(purse.brand)
+      ) {
+        continue;
+      }
```

Comment:
consider distinguishing these conditions:
```
if (!Array.isArray(purse.balance.value)) {
  console.debug('skipping non-set amount', purse.balance.value);
}
if (!agoricBrands.has(purse.brand)) {
  console.debug('skipping unknown brand', purse.brand);
```
Likely to pay off debugging someday. And in the meantime it's code documentation.

Reply:
Done